packages/core/src/libraries/user.ts

```diff
@@ -88,7 +93,17 @@ export const insertUser: typeof insertUserQuery = async ({ roleNames, ...rest })
     }
   }
 
-  return insertUserQuery({ roleNames: computedRoleNames, ...rest });
+  const user = await insertUserQuery(rest);
```

Comment:
these queries should be in a transaction?

Reply:
Good call, but will need to change the existing query functions to receive outside "connection", created LOG-4294 to enhance.